### ISSUE TITLE
Date picker calendar has suboptimal UX

### ISSUE THREAD

**Issue by andrewler** [metabase/metabase#35352]
### Describe the bug

While using the date picker calendar, some months do not display the complete set of "week rows" for the current month. I have to move to the next or previous month to select the date of the current month.

For example when November 2023 is selected I can only see November 5 thru December 2. Therefore if I want to select November 3, I must select October as the month.

The strange thing is I can see that the calendar UI supports displaying up to 5 "week rows" but November only shows a maximum of 4 "week rows." Expanding on this, December 2023 would require 6 "week rows" to display the entire month to the user.

![meta-datepicker-ui](https://github.com/metabase/metabase/assets/76549682/4d3aa5d3-e585-493b-a5ab-06788152cb16)

### To Reproduce

1. Create a dashboard with a date picker
2. In the date picker go to November 2023
3. Try to select November 3rd as the date. You can't.
4. You must go to October 2023 to select November 3rd 2023.

### Expected behavior

I should be able to see all of the relevant "week rows" of the currently select month so I don't have to go to a different month (+/- 1 month) to select the date I want

### Information about your Metabase installation

```JSON
{
  "browser-info": {
    "language": "en-US",
    "platform": "Win32",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.18+10",
    "java.vendor": "Eclipse Adoptium",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "11.0.18",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.18+10",
    "os.name": "Linux",
    "os.version": "5.11.0-41-generic",
    "user.language": "en",
    "user.timezone": "Asia/Seoul"
  },
  "metabase-info": {
    "databases": [
      "mysql",
      "h2"
    ],
    "hosting-env": "unknown",
    "application-database": "mysql",
    "application-database-details": {
      "database": {
        "name": "MySQL",
        "version": "5.7.35"
      },
      "jdbc-driver": {
        "name": "MariaDB Connector/J",
        "version": "2.7.6"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2023-03-28",
      "tag": "v0.46.0",
      "branch": "release-x.46.x",
      "hash": "0ca7df3"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

### Severity

Nitpicking

**Comment by paoliniluis** [comment]
Please upgrade to the latest version and check if this is still the case

**Comment by andrewler** [comment]
> Please upgrade to the latest version and check if this is still the case @andrewler

@paoliniluis Issue is resolved after upgrading metabase instance to 0.47.7 

![image](https://github.com/metabase/metabase/assets/76549682/30569741-5628-4a0e-8d22-258862a442fe)
